generators/server/templates/src/main/java/package/config/_WebConfigurer.java

```diff
@@ -116,11 +116,14 @@ public void customize(ConfigurableEmbeddedServletContainer container) {
         // CloudFoundry issue, see https://github.com/cloudfoundry/gorouter/issues/64
         mappings.add("json", "text/html;charset=utf-8");
         container.setMimeMappings(mappings);<% if (!skipClient) { %>
-        // Set document root
-        if (env.acceptsProfiles(Constants.SPRING_PROFILE_PRODUCTION)) {
-            container.setDocumentRoot(new File("<%= CLIENT_DIST_DIR %>"));
-        } else if (env.acceptsProfiles(Constants.SPRING_PROFILE_DEVELOPMENT)) {
-            container.setDocumentRoot(new File("<%= CLIENT_MAIN_SRC_DIR %>"));
+
+        // Set document root if we're not running from a jar/war
+        if (getClass().getProtectionDomain().getCodeSource().getLocation().getProtocol().equals("file")) {
```

Comment:
Well sure, if you like that better... The reason for doing that doesn't apply here, though, the `getProtocol` is never going to return `null`.